screenpipe-server/src/migrations/20241029015040_audio_transcription_audio_chunk_id_nullable.sql

```diff
@@ -0,0 +1,29 @@
+ALTER TABLE audio_transcriptions RENAME TO audio_transcriptions_old;
+
+CREATE TABLE audio_transcriptions (
+    id INTEGER PRIMARY KEY AUTOINCREMENT,
+    audio_chunk_id INTEGER,
+    transcription TEXT NOT NULL,
+    offset_index INTEGER NOT NULL,
+    timestamp TIMESTAMP NOT NULL,
+    transcription_engine TEXT NOT NULL DEFAULT 'Whisper',
+    device TEXT NOT NULL DEFAULT '',
+    is_input_device BOOLEAN NOT NULL DEFAULT TRUE,
+    FOREIGN KEY (audio_chunk_id) REFERENCES audio_chunks(id)
```

Comment:
is it nullable?

kinda scary mutation, i have 120 gb of screenpipe data, can it take a while?

Reply:
`audio_chunk_id` isn't currently nullable. Yeah this is super non-ideal but seems like the only way to make the column nullable in SQLite. Not sure how long it might take but I doubt too long considering it is just text data. Other options I can think of are removing the foreign key constraint or creating a dummy (or multiple dummy) audio_chunks

Reply:
Never mind same issue removing the foreign key constraint. Dummy audio_chunks is the only solution I can think of.

Reply:
maybe something clean is to remove foreign key and be optional string and in the audio_chunks table add a foreign key reference to audio_transcriptions 

?

Reply:
why cant we remove foreign key?

Reply:
You have to do the same process of creating a new table and copying over. The audio_chunks `file_path` is also not nullable so will need to put in "no_path" or just an empty string

Reply:
lets jsut do dummy chunk with empty file path